tests/NLog.UnitTests/LoggerTests.cs

```diff
@@ -1813,6 +1813,34 @@ public void SingleTargetMessageFormatOptimizationTest()
             AssertDebugLastMessage("target2", "DualTarget|Hello World");
         }
 
+        [Theory]
+        [InlineData(null, "OrderId", "@Client")]
+        [InlineData(true, "OrderId", "@Client")]
+        [InlineData(null, "0", "@Client")] //fails: gives "Process order {0} for {@Client}"
```

Comment:
Doubting if this is a bug, at least it's unwanted

Reply:
Works as expected, not valid string.Format input, so it just returns the raw unformatted message. Unless you want to have fallback on exception?